### PR TITLE
[fix][broker] Fix broker LoadBalance uneffective

### LINKED ISSUE #15313: Broker LoadBalance uneffective because broker ResourceUsage is always 0
**Describe the bug**

leader broker calculate broker resource usage(e.g. cpu,memory,directMemory) is 0 when do loadbalance.

loadbalance log:
```
2022-04-25 12:58:18,285Z [pulsar-load-manager-1-1] WARN  org.apache.pulsar.broker.loadbalance.impl.ThresholdShedder - average max resource usage is 0

```

broker configs about loadbalance
```
loadBalancerEnabled=true

loadBalancerLoadSheddingStrategy=org.apache.pulsar.broker.loadbalance.impl.ThresholdShedder

loadBalancerCPUResourceWeight=1.0
loadBalancerMemoryResourceWeight=1.0
loadBalancerDirectMemoryResourceWeight=1.0
loadBalancerBandwithInResourceWeight=1.0
loadBalancerBandwithOutResourceWeight=1.0
```

broker data in zookeeper path:  /loadbalance/brokers/
```
{"webServiceUrl":"http://10.244.2.23:8080","webServiceUrlTls":"https://10.244.2.23:8081","pulsarServiceUrlTls":"pulsar+ssl://10.244.2.23:6651","persistentTopicsEnabled":true,"nonPersistentTopicsEnabled":false,"cpu":{"usage":3.1577712104798255,"limit":100.0},"memory":{"usage":609.6469955444336,"limit":1228.0},"directMemory":{"usage":32.0,"limit":1228.0},"bandwidthIn":{"usage":0.0,"limit":0.0},"bandwidthOut":{"usage":0.0,"limit":0.0},"msgThroughputIn":0.0,"msgThroughputOut":0.0,"msgRateIn":0.0,"msgRateOut":0.0,"lastUpdate":1650886425227,"lastStats":{"pulsar/pulsar/10.244.2.23:8080/0x00000000_0xffffffff":{"msgRateIn":0.0,"msgThroughputIn":0.0,"msgRateOut":0.0,"msgThroughputOut":0.0,"consumerCount":0,"producerCount":0,"topics":1,"cacheSize":0}},"numTopics":1,"numBundles":1,"numConsumers":0,"numProducers":0,"bundles":["pulsar/pulsar/10.244.2.23:8080/0x00000000_0xffffffff"],"lastBundleGains":[],"lastBundleLosses":[],"brokerVersionString":"2.11.0-hw-0.0.4-SNAPSHOT","protocols":{},"advertisedListeners":{},"bundleStats":{"pulsar/pulsar/10.244.2.23:8080/0x00000000_0xffffffff":{"msgRateIn":0.0,"msgThroughputIn":0.0,"msgRateOut":0.0,"msgThroughputOut":0.0,"consumerCount":0,"producerCount":0,"topics":1,"cacheSize":0}},"maxResourceUsage":0.49645519256591797,"loadReportType":"LocalBrokerData"}
```

And we found that problem happens after [pr#13639](https://github.com/apache/pulsar/pull/13639)

The reason is zk json data deserialize to pojo `org.apache.pulsar.policies.data.loadbalancer.ResourceUsage` will not use constructor `ResourceUsage(double usage, double limit)`， so that percentUsage is always 0
```
public ResourceUsage(double usage, double limit) {
        this.usage = usage;
        this.limit = limit;
        float proportion = 0;
        if (limit > 0) {
            proportion = ((float) usage) / ((float) limit);
        }
        percentUsage = proportion * 100;
    }
```

Broker 2.10.0

### PR DESCRIPTION
**Pull request by fu-turer**
Fixes #15313 

### Motivation

fix ResourceUsage percentUsage is always 0 when construct by json serialize

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [x] `no-need-doc` 
(Please explain why)

### CODE REVIEW

**Comment by mattisonchao** [comment]
Hi @fu-turer 
Could we add the test to ensure this change?

**Comment by fu-turer** [comment]
> Hi @fu-turer Could we add the test to ensure this change?

@mattisonchao  sure.

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/policies/data/loadbalancer/ResourceUsage.java [hunk 1]
```diff
@@ -27,17 +27,10 @@
 public class ResourceUsage {
     public final double usage;
     public final double limit;
-    @EqualsAndHashCode.Exclude
```
Comment (michaeljmarshall) [inline pulsar-client-admin-api/src/main/java/org/apache/pulsar/policies/data/loadbalancer/ResourceUsage.java:30]: Nit: remove the import line for this annotation.

File: pulsar-common/src/test/java/org/apache/pulsar/policies/data/loadbalancer/LocalBrokerDataTest.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        Gson gson = new Gson();
+        LocalBrokerData localBrokerData = gson.fromJson(data, LocalBrokerData.class);
+        Assert.assertEquals(localBrokerData.getMemory().limit, 1228.0d, 0.0001f);
+        Assert.assertEquals(localBrokerData.getMemory().usage, 614.0d, 0.0001f);
+        Assert.assertEquals(localBrokerData.getMemory().percentUsage(), ((float) localBrokerData.getCpu().usage) / ((float) localBrokerData.getCpu().limit) * 100, 0.0001f);
```
Comment (michaeljmarshall) [inline pulsar-common/src/test/java/org/apache/pulsar/policies/data/loadbalancer/LocalBrokerDataTest.java:34]:
```suggestion
        Assert.assertEquals(localBrokerData.getMemory().percentUsage(), ((float) localBrokerData.getMemory().usage) / ((float) localBrokerData.getMemory().limit) * 100, 0.0001f);
```

**Review by michaeljmarshall — CHANGES_REQUESTED** [review]
LGTM. Great catch @fu-turer! I have two minor comments you'll need to address before we can merge this.

I didn't know that Gson modified final fields when deserializing JSON objects. Also, I'm a bit surprised that we don't have an integration covering this code path. Your unit test looks good, but I would have thought we'd have functional load balancing tests.

File: pulsar-client-admin-api/src/main/java/org/apache/pulsar/policies/data/loadbalancer/ResourceUsage.java (hunk 1 above)
Comment (fu-turer) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/policies/data/loadbalancer/ResourceUsage.java:30]: there's no import for `@EqualsAndHashCode.Exclude`

Comment (michaeljmarshall) [reply pulsar-client-admin-api/src/main/java/org/apache/pulsar/policies/data/loadbalancer/ResourceUsage.java:30]: Oh, right, sorry about that.

**Review by michaeljmarshall — APPROVED** [review]

**Review by lhotari — APPROVED** [review]

**Pull request merged**